src/core/Akka/Actor/CoordinatedShutdown.cs

```diff
@@ -653,13 +653,15 @@ internal static void InitPhaseActorSystemTerminate(ActorSystem system, Config co
 
                     if (terminateActorSystem)
                     {
-                        system.FinalTerminate();
-                        return system.Terminate().ContinueWith(tr =>
+                        return system.FinalTerminate().ContinueWith(tr =>
                         {
                             if (exitClr && !coord._runningClrHook)
                             {
                                 Environment.Exit(0);
                             }
+
+                            coord.System = null;
```

Comment:
Clear the reference to the `ActorSystem` to prevent a circular reference, preventing the `ActorSystem` from being garbage collected afterward.